File: src/main/java/diff/patch/accessors/FieldSetter.java
package diff.patch.accessors;

import diff.patch.Setter;

import java.lang.reflect.Field;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class FieldSetter<Type> implements Setter<Type> {
    final Field field;
    final Object target;

    public FieldSetter(Object target, Field field) {
        this.field = field;
        this.target = target;
        field.setAccessible(true);
    }

    @Override
    public void set(Type value) {
        try {
            field.set(target, value);
        } catch (IllegalArgumentException | IllegalAccessException ignored) {
            ignored.printStackTrace();
        }
    }
    
}

File: src/main/kotlin/diff/patch/accessors/FieldSetter.kt
package diff.patch.accessors

import diff.patch.Setter

import java.lang.reflect.Field

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class FieldSetter<Type: Any>(internal val target: Any, internal val field: Field) : Setter<Type> {

    init {
        field.isAccessible = true
    }

    override fun set(value: Type) {
        try {
            field.set(target, value)
        } catch (ignored: IllegalArgumentException) {
            ignored.printStackTrace()
        } catch (ignored: IllegalAccessException) {
            ignored.printStackTrace()
        }

    }

}
